opflexagent/vpp_plug_firewall.py

```diff
@@ -0,0 +1,88 @@
+                if os_vif_util._is_firewall_required(vif) or \
+                    vif.is_hybrid_plug_enabled():
+                    profile.bridge_name = \
+                        os_vif_util._get_hybrid_bridge_name(vif)
+                    profile.hybrid_plug = True
+                else:
+                    profile.hybrid_plug = False
+                    if vif["network"]["bridge"] is not None:
+                        profile.bridge_name = vif["network"]["bridge"]
+            else:
+                profile = objects.vif.VIFPortProfileFPBridge()
+                if vif["network"]["bridge"] is not None:
+                    profile.bridge_name = vif["network"]["bridge"]
+            obj = os_vif_util._get_vif_instance(vif, objects.vif.VIFVHostUser,
+                            plugin="vhostuser_fp",
+                            vif_name=os_vif_util._get_vif_name(vif),
+                            port_profile=profile)
+            os_vif_util._set_vhostuser_settings(vif, obj)
+            return obj
+        elif vif['details'].get(model.VIF_DETAILS_VHOSTUSER_OVS_PLUG, False):
+            profile = objects.vif.VIFPortProfileOpenVSwitch(
+                interface_id=vif.get('ovs_interfaceid') or vif['id'],
+                datapath_type=vif['details'].get(
+                    model.VIF_DETAILS_OVS_DATAPATH_TYPE))
+            vif_name = ('vhu' + vif['id'])[:model.NIC_NAME_LEN]
+            obj = os_vif_util._get_vif_instance(vif, objects.vif.VIFVHostUser,
+                                    port_profile=profile, plugin="ovs",
+                                    vif_name=vif_name)
+            if vif["network"]["bridge"] is not None:
+                obj.bridge_name = vif["network"]["bridge"]
+                os_vif_util._set_vhostuser_settings(vif, obj)
+            return obj
+        elif vif['details'].get(
+            model.VIF_DETAILS_VHOSTUSER_VROUTER_PLUG, False):
+            obj = os_vif_util._get_vif_instance(vif, objects.vif.VIFVHostUser,
+                                    plugin="contrail_vrouter",
+                                    vif_name=os_vif_util._get_vif_name(vif))
+            os_vif_util._set_vhostuser_settings(vif, obj)
+            return obj
+        elif vif['details'].get("vhostuser_vpp_plug", False):
+            profile = objects.vif.VIFPortProfileOpenVSwitch(
+                interface_id=vif.get('ovs_interfaceid') or vif['id'],
+                datapath_type=model.VIF_DETAILS_OVS_DATAPATH_NETDEV)
+            vif_name = ('vhu' + vif['id'])[:model.NIC_NAME_LEN]
+            obj = os_vif_util._get_vif_instance(vif, objects.vif.VIFVHostUser,
+                                    port_profile=profile, plugin="vpp",
+                                    vif_name=os_vif_util._get_vif_name(vif))
+            os_vif_util._set_vhostuser_settings(vif, obj)
+            return obj
+        else:
+            raise NotImplementedError()
+
+    vpp_nova_to_osvif_vif_vhostuser = \
```

Comment:
ditto

Reply:
Naming seems right.. this is monkey patching the corresponding os_vif function
_nova_to_osvif_vif_vhostuser

Reply:
I was referring to the usage of brackets instead of slash for line continuation